scripts/mergehex.py

```diff
@@ -18,11 +19,18 @@ def merge_hex_files(output, input_hex_files):
     for hex_file_path in input_hex_files:
         to_merge = IntelHex(hex_file_path)
 
-        # Since 'arm-none-eabi-objcopy' incorrectly inserts record type '03 - Start Segment Address', we need to remove
-        # the start_addr to avoid conflicts when merging.
+        # Since 'arm-none-eabi-objcopy' incorrectly inserts record
+        # type '03 - Start Segment Address', we need to remove the
+        # start_addr to avoid conflicts when merging.
         to_merge.start_addr = None
 
-        ih.merge(to_merge)
+        try:
+            ih.merge(to_merge)
+        except AddressOverlapError as e:
+            raise AddressOverlapError("{} has merge issues".format(hex_file_path))
+
+        print("Merged {}".format(hex_file_path))
```

Comment:
Should this be here?

Reply:
I'd say so.

You could argue that it should be in a different commit though.

Before:

```
[394/395] Linking C executable zephyr/zephyr.elf
[395/395] Generating merged.hex

```
After:

```
[394/395] Linking C executable zephyr/zephyr.elf
[395/395] Generating merged.hex
Merged /home/sebo/ncs/zephyr/samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/b/zephyr/subsys/boot/mcuboot/mcuboot/zephyr/zephyr.hex
Merged signed.hex

```